library/libwaku.h

```diff
@@ -15,45 +15,73 @@
 extern "C" {
 #endif
 
-typedef void (*WakuCallBack) (const char* msg, size_t len_0);
+typedef void (*WakuCallBack) (int callerRet, const char* msg, size_t len);
+
+// Initializes the waku library and returns a pointer to the Context.
+void* waku_init(WakuCallBack callback,
+                void* userData);
 
 // Creates a new instance of the waku node.
 // Sets up the waku node from the given configuration.
-int waku_new(const char* configJson, WakuCallBack onErrCb);
+int waku_new(void* ctx,
```

Comment:
for the context, a useful trick is to define an opaque type for it (`typedef struct WakuContext WakuContext;`) and pass around `WakuContext*` instead of `void*` - this documents intent and gives some rudimentary type safety (even if it's limited in C).